### ISSUE TITLE
Linking to a remote project fails using supabase cli

### ISSUE THREAD

**Issue by Evaldas-B** [supabase/cli#646]
# Bug report

## Describe the bug

Linking to a newly created supabase project fails using `npx supabase link --project-ref $projectRef -p $password` but is successful when debug flag is included: `npx supabase link --project-ref $projectRef -p password --debug`

## To Reproduce

1. Create a new project from supabase.com or CLI
2. Run `npx supabase link --project-ref $projectRef -p password`
3. Linking is failed error is thrown:

`Error: failed to connect to host=db.PROJECT-REF.supabase.co user=postgres database=postgres: dial error (dial tcp 127.0.0.1:6543: connectex: No connection could be made because the target machine actively refused it.)
Try rerunning the command with --debug to troubleshoot the error.`

## Expected behavior
Finished supabase link message should be displayed and .temp/project-ref file should be created

## Screenshots

![image](https://user-images.githubusercontent.com/53705834/204362974-cce48915-f714-4ede-bd13-715b8851be52.png)

## System information

- Supabase CLI: 1.16.2

**Comment by sweatybridge** [comment]
It seems like your project domain name is resolving to localhost. Could you check the output of the following PowerShell command?

```
PS C:\> Resolve-DnsName -Name db.PROJECT-REF.supabase.co
```

**Comment by princemaple** [comment]
Hi, I'm getting the same error.

DNS resolution seems to be fine. It points to an EC2 domain.

**Comment by Evaldas-B** [comment]
> It seems like your project domain name is resolving to localhost. Could you check the output of the following PowerShell command?
> 
> ```
> PS C:\> Resolve-DnsName -Name db.PROJECT-REF.supabase.co
> ```

Can confirm that it points to `ec2-35-158-234-7.eu-central-1.compute.amazonaws.com`

![image](https://user-images.githubusercontent.com/53705834/204473139-69d2b466-e225-433b-bb02-8b834d69be4c.png)

**Comment by princemaple** [comment]
@sweatybridge is it normal that my ping to the db url times out?

Is it a possible cause: after a project is paused then resumed, the db is at a different machine but the cname is not updated?

**Comment by sweatybridge** [comment]
Thanks for letting us know how you resolved this issue. I've pinged our infra team to check if they have a better explanation.

**Comment by princemaple** [comment]
Still can't run `db remote commit`, got the same error. Would really appreciate it someone could provide some insight.

**Comment by sweatybridge** [comment]
Could you contact support@supabase.io with your project ref? I can help take a look

**Comment by princemaple** [comment]
Thanks. Sent.

After sending the email, I found another interesting thing! With `supabase --debug link ...` it worked 🤣 Without `--debug` it reports the previously mentioned error. Very funny. Same with many other commands, e.g. `migration list`.

`db remote commit` still failed but did manage to connect the db. Logs:

```
❯ : supabase --debug db remote commit
2022/12/03 13:19:56 Connect Start: tcp 54.206.65.220:6543
2022/12/03 13:19:56 Connect Done: tcp 54.206.65.220:6543
2022/12/03 13:19:56 PG Send: {"Type":"StartupMessage","ProtocolVersion":196608,"Parameters":{"database":"postgres","user":"postgres"}}
2022/12/03 13:19:56 PG Recv: {"Type":"AuthenticationSASL","AuthMechanisms":["SCRAM-SHA-256"]}
2022/12/03 13:19:56 PG Send: {"Type":"SASLInitialResponse","AuthMechanism":"SCRAM-SHA-256","Data":"n,,n=,r=Suez6fkkUnoj18oDPAy6sA81"}
2022/12/03 13:19:56 PG Recv: {"Type":"AuthenticationSASLContinue","Data":"r=Suez6fkkUnoj18oDPAy6sA817kSlJ+GQV4ltWamai3mJdh6i,s=KBEh64Byw3EWCjAMuWkVIA==,i=4096"}
2022/12/03 13:19:56 PG Send: {"Type":"SASLResponse","Data":"c=biws,r=Suez6fkkUnoj18oDPAy6sA817kSlJ+GQV4ltWamai3mJdh6i,p=baDk0GUi145UZcNjNq7oYJQ5/3VDvnikS79R6SuE8s8="}
2022/12/03 13:19:56 PG Recv: {"Type":"AuthenticationSASLFinal","Data":"v=PCuluxzHqq3Z1sejS8BoYkBMXYFz4z8jKqNHKyGQ/ps="}
2022/12/03 13:19:56 PG Recv: {"Type":"AuthenticationOK"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"default_transaction_read_only","Value":"off"}   
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"in_hot_standby","Value":"off"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"integer_datetimes","Value":"on"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"IntervalStyle","Value":"postgres"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"is_superuser","Value":"off"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"server_encoding","Value":"UTF8"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"server_version","Value":"14.1"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"client_encoding","Value":"UTF8"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"DateStyle","Value":"ISO, MDY"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"TimeZone","Value":"UTC"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"standard_conforming_strings","Value":"on"}
2022/12/03 13:19:56 PG Recv: {"Type":"ParameterStatus","Name":"application_name","Value":""}
2022/12/03 13:19:56 PG Recv: {"Type":"BackendKeyData","ProcessID":3950203809,"SecretKey":3651288543} 
2022/12/03 13:19:56 PG Recv: {"Type":"ReadyForQuery","TxStatus":"I"}
2022/12/03 13:19:56 PG Send: {"Type":"Query","String":"SELECT version FROM supabase_migrations.schema_migrations ORDER BY version"}
2022/12/03 13:19:56 PG Recv: {"Type":"RowDescription","Fields":[{"Name":"version","TableOID":26282,"TableAttributeNumber":1,"DataTypeOID":25,"DataTypeSize":-1,"TypeModifier":-1,"Format":0}]}
2022/12/03 13:19:56 PG Recv: {"Type":"DataRow","Values":[{"text":"20220829125658"}]}
2022/12/03 13:19:56 PG Recv: {"Type":"CommandComplete","CommandTag":"SELECT 1"}
2022/12/03 13:19:56 PG Recv: {"Type":"ReadyForQuery","TxStatus":"I"} 
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:56 Recv First Byte
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Sent Header: Content-Length [310]
2022/12/03 13:19:56 Sent Header: Content-Type [application/json]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:56 Recv First Byte
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:56 Recv First Byte
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:56 Recv First Byte
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Sent Header: Content-Length [1735]
2022/12/03 13:19:56 Sent Header: Content-Type [application/json]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:56 Recv First Byte
2022/12/03 13:19:56 Sent Header: Host [docker]
2022/12/03 13:19:56 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:56 Sent Header: Content-Length [0]
2022/12/03 13:19:56 Sent Header: Content-Type [text/plain]
2022/12/03 13:19:56 Send Done
2022/12/03 13:19:57 Recv First Byte
2022/12/03 13:19:57 Sent Header: Host [docker]
2022/12/03 13:19:57 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:57 Sent Header: Content-Length [4462]
2022/12/03 13:19:57 Sent Header: Content-Type [application/json]
2022/12/03 13:19:57 Send Done
2022/12/03 13:19:57 Recv First Byte
2022/12/03 13:19:58 Sent Header: Host [docker]
2022/12/03 13:19:58 Sent Header: User-Agent [Go-http-client/1.1]
2022/12/03 13:19:58 Sent Header: Content-Length [101269]
2022/12/03 13:19:58 Sent Header: Content-Type [application/json]
2022/12/03 13:19:58 Send Done
2022/12/03 13:19:58 Recv First Byte
2022/12/03 13:19:58 PG Send: {"Type":"Terminate"}
Error: error creating shadow database
```

**Comment by princemaple** [comment]
@Evaldas-B You probably wanna try `--debug`, see my previous comment.

**Comment by sweatybridge** [comment]
Thanks for helping to investigate. I can confirm there's an issue with remote commit if you are running it for second time. 

One workaround is to delete entries in `supabase_migrations.schema_migrations` table and run remote commit again.

**Comment by sweatybridge** [comment]
The root cause with db host resolving inconsistently across platforms was identified. We have released a fix for windows in 1.17.5.

**Comment by princemaple** [comment]
Thank you. Will give this a go tomorrow!
